R/dominantTaxa.R

```diff
@@ -100,9 +106,56 @@ setMethod("perSampleDominantFeatures", signature = c(x = "SummarizedExperiment")
         idx <- as.list(apply(t(mat) == colMaxs(mat),1L,which))
         # Get rownames based on indices
         taxa <- rownames(mat)[unlist(idx)]
+        
         # If multiple dominant taxa were found, names contain taxa in addition to 
         # sample name. Names are converted so that they include only sample names.
         names(taxa) <- rep( names(idx), times = lengths(idx) )
+        
+        # If individual sample contains multiple dominant taxa (they have equal counts) and if 
+        # complete is FALSE, the an arbitrarily chosen dominant taxa is returned
+        if( length(taxa) > nrow(colData(x))){
+            # Store order
+            order <- unique(names(taxa))
+            # there are multiple dominant taxa in one sample (counts are equal), length
+            # of dominant is greater than rows in colData.
+            taxa <- split(taxa, rep(names(taxa), lengths(taxa)) )
+            # Order the data
+            taxa <- taxa[order]
+            
+            # If complete is set FALSE, and there are multiple dominant taxa, one of them is arbitrarily chosen
+            if( complete == FALSE) {
+                # Randomly choose one element from each sublist
+                taxa <- lapply(taxa, function(item) {
+                                if (length(item) > 1) {
+                                # If it's a list with more than one element, choose one randomly
+                                    return(sample(item, 1))
+                                } else {
+                                    return(item)
+                                }
```

Comment:
You could use sample() everytime to simplify the code. If there is only one element, then the sample is always the same